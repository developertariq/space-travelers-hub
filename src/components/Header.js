import React from 'react';
import logo from '../images/planet.png';
import Navbar from './Navbar';

const Header = () => (
  <header className="App-header" data-testid="header">
    <img src={logo} alt="Planet logo" className="App-logo" />
    <h1 className="title" data-testid="site-title">
      Space Travelers&apos; Hub
    </h1>
    <Navbar />
  </header>
);

export default Header;
