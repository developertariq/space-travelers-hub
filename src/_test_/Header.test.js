import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import Header from '../components/Header';

describe('Header', () => {
  test('renders correctly', () => {
    const tree = renderer.create(
      <Router>
        <Header />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the correct header', () => {
    const { getByTestId } = render(
      <Router>
        <Header />
      </Router>,
    );
    const header = getByTestId('header');
    expect(header).toBeInTheDocument();
  });

  test('renders the correct title', () => {
    const { getByTestId } = render(
      <Router>
        <Header />
      </Router>,
    );
    const siteTitle = getByTestId('site-title');
    expect(siteTitle).toBeInTheDocument();
  });

  test('renders the correct navbar', () => {
    const { getByTestId } = render(
      <Router>
        <Header />
      </Router>,
    );
    const navbar = getByTestId('navbar');
    expect(navbar).toBeInTheDocument();
  });
});
