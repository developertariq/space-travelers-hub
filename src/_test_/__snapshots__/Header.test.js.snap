// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header section correctly 1`] = `
<header
  className="App-header"
  data-testid="header"
>
  <img
    alt="Planet logo"
    className="App-logo"
    src="planet.png"
  />
  <h1
    className="title"
    data-testid="site-title"
  >
    Space Travelers' Hub
  </h1>
  <nav
    className="NavBar"
    data-testid="navbar"
  >
    <ul
      className="MenuNav"
    >
      <li>
        <a
          activeClassName="active-link"
          aria-current="page"
          className="active"
          exact={true}
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          activeClassName="active-link"
          className=""
          exact={true}
          href="/Missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          activeClassName="active-link"
          className=""
          exact={true}
          href="/MyProfile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Header renders correctly 1`] = `
<header
  className="App-header"
  data-testid="header"
>
  <img
    alt="Planet logo"
    className="App-logo"
    src="planet.png"
  />
  <h1
    className="title"
    data-testid="site-title"
  >
    Space Travelers' Hub
  </h1>
  <nav
    className="NavBar"
    data-testid="navbar"
  >
    <ul
      className="MenuNav"
    >
      <li>
        <a
          activeClassName="active-link"
          aria-current="page"
          className="active"
          exact={true}
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          activeClassName="active-link"
          className=""
          exact={true}
          href="/Missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          activeClassName="active-link"
          className=""
          exact={true}
          href="/MyProfile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
